src/Order.Data/OrderContext.cs

```diff
@@ -1,160 +1,208 @@
+                    .HasColumnType("binary(16)");
+
+                entity.HasOne(d => d.Service)
+                    .WithMany(p => p.OrderProduct)
+                    .HasForeignKey(d => d.ServiceId)
+                    .OnDelete(DeleteBehavior.ClientSetNull)
+                    .HasConstraintName("order_service_opfk_1");
+            });
+
+            modelBuilder.Entity<OrderService>(entity =>
+            {
+                entity.ToTable("order_service");
+
+                entity.Property(e => e.Id).HasColumnType("binary(16)");
+
+                entity.Property(e => e.Name)
+                    .IsRequired()
+                    .HasMaxLength(100)
+                    .IsUnicode(false);
+            });
+
+            modelBuilder.Entity<OrderStatus>(entity =>
+            {
+                entity.ToTable("order_status");
+
+                entity.HasIndex(e => e.Name)
+                    .HasName("Status");
+
+                entity.Property(e => e.Id).HasColumnType("binary(16)");
+
+                entity.Property(e => e.Name)
+                    .IsRequired()
+                    .HasMaxLength(20)
+                    .IsUnicode(false);
+            });
+        }
+
+        /// <summary>
+        /// This configuration is required because SQLite provider cannot apply aggregate operators in queries 
+        /// for navigation properties with decimal and DateTimeOffset type.
+        /// SQLite provider is used only for tests, so this change applies only for tests.
+        /// </summary>
+        private void ConfigureConversionsForTestDatabase(ModelBuilder modelBuilder)
+        {
+            if (Database.ProviderName == "Microsoft.EntityFrameworkCore.Sqlite")
+            {
+                foreach (var entityType in modelBuilder.Model.GetEntityTypes())
+                {
+                    var decimalProperties = entityType.ClrType.GetProperties()
+                        .Where(p => p.PropertyType == typeof(decimal));
+                    var dateTimeProperties = entityType.ClrType.GetProperties()
+                        .Where(p => p.PropertyType == typeof(DateTimeOffset));
+
+                    foreach (var property in decimalProperties)
+                    {
+                        modelBuilder.Entity(entityType.Name).Property(property.Name).HasConversion<double>();
+                    }
+
+                    foreach (var property in dateTimeProperties)
+                    {
+                        modelBuilder.Entity(entityType.Name).Property(property.Name)
+                            .HasConversion(new DateTimeOffsetToBinaryConverter());
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sqlite is used only for in memory database in tests.
+        /// Todo: This might be replaced with proper conversion in method above.
+        /// </summary>
+        /// <returns></returns>
+        public bool IsInMemoryDatabase()
```

Comment:
I think we could get rid of this method and requirements for different comparisons of Guid Ids for tests and production code with the implementation of conversion in the method above. But I was not able to confirm or implement it in time.